pytest/README.md

```diff
@@ -0,0 +1,161 @@
+# LISAv3 via pytest-lisa
+
+[Pytest](https://docs.pytest.org/en/stable/) is an [incredibly
+popular](https://docs.pytest.org/en/stable/talks.html) MIT licensed open source
+Python testing framework. It has a thriving community and plugin framework, with
+[over 750 plugins](https://plugincompat.herokuapp.com/). There is even a YAML
+example of writing a Domain Specific Language
+[DSL](https://docs.pytest.org/en/stable/example/nonpython.html#yaml-plugin) for
+specifying tests. Instead of writing yet another test framework, LISAv3 could be
+written as pytest-lisa, a [plugin for
+Pytest](https://docs.pytest.org/en/stable/writing_plugins.html) which implements
+our requirements. In fact, most of Pytest itself is implemented via [built-in
+plugins](https://docs.pytest.org/en/stable/plugins.html), providing us with a
+lot to leverage.
+
+The [fundamental features](https://www.youtube.com/watch?v=CMuSn9cofbI) of
+Pytest match our needs very well:
+
+* Automatic test discovery, no boiler-plate test code
+* Useful information when a test fails (assertions are introspected)
+* Test parameterization
+* Modular setup/teardown via fixtures
+* Customizable (as detailed above)
+
+So all the logic for discovering, running, skipping based on requirements, and
+reporting the tests is already written and maintained by the greater open source
+community, leaving us to focus on the hard and unique problem: creating an API
+to launch the necessary nodes. It would also allow us the space to abstract the
+installation of tools required by tests. In this way, LISAv3 could solve the
+difficulties we have at hand without creating yet another unit test framework.
+
+## Design
+
+### pytest-mark
+
+The [pytest-mark](https://docs.pytest.org/en/stable/mark.html) already provides
```

Comment:
Can it select test cases bases on the marks?

Reply:
Yes, that is the main intention of marks. It's also incredibly flexible so we can make it do whatever we want, while still leveraging everything underneath.